hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -550,6 +674,28 @@ public final Stream<HoodieBaseFile> getLatestBaseFilesBeforeOrOn(String partitio
     }
   }
 
+  @Override
+  public final Map<String, Stream<HoodieBaseFile>> getAllLatestBaseFilesBeforeOrOn(String maxCommitTime) {
+    try {
+      readLock.lock();
+
+      List<String> formattedPartitionList = ensureAllPartitionsLoadedCorrectly();
+      return formattedPartitionList.stream().collect(Collectors.toMap(
+          Function.identity(),
+          partitionPath -> fetchAllStoredFileGroups(partitionPath)
+              .filter(fileGroup -> !isFileGroupReplacedBeforeOrOn(fileGroup.getFileGroupId(), maxCommitTime))
```

Comment:
is it possible to reuse `getLatestBaseFilesBeforeOrOn(String partitionStr, String maxCommitTime)`  to avoid code duplication. just incase we make some bug fix in the other method, might forget to fix here.

Reply:
We can extract the core logic of filtering file groups.  We cannot directly reuse `getLatestBaseFilesBeforeOrOn()` because it triggers metadata table lookup per partition, which this PR is trying to fix.